### PR TITLE
Add button to reset passcode

### PR DESCRIPTION
**Pull request by DerGut**
I did some work on #63:
The PR adds a new button to the `<TeamDisplayCard />`.

Clicking it will generate a new passcode, patch the teams juice-shop deployment with the new hash and return the passcode to the user. The passcode will then be displayed via the old but modified `<PasscodeDisplayCard />`.

Looking forward to your feedback!

Screenshot 2020-10-08 at 15 40 56

### CODE REVIEW

**Comment by J12934** [comment]
Wow nice 🚀
Will have a closer look later 👍

**Review by J12934 — CHANGES_REQUESTED** [review]
Seems to be working great 👏
If marked some minor spots where you are using sync code where it could be written asynchronously.


One thing we could potentially change to make this more flexible is to change the `resetPasscode` endpoint to accept the team name as a request argument. This way we could also use the endpoint on the admin page so that an admin can change the passcode of a team.

But this should be flexible enough for now 👍

File: juice-balancer/src/teams/teams.js [hunk 1]
```diff
@@ -153,15 +154,20 @@ async function checkIfMaxJuiceShopInstancesIsReached(req, res, next) {
   }
 }
 
+function generatePasscode() {
+  const passcode = cryptoRandomString({ length: 8 }).toUpperCase();
+  const hash = bcrypt.hashSync(passcode, BCRYPT_ROUNDS);
+  return { passcode, hash };
+}
```
[the remark this answers is not in the document]
Comment (DerGut) [reply juice-balancer/src/teams/teams.js:161]: Great! 👍  I thought the sync functions would simply await the call behind the scenes. Apparently there is more to it...

**Review by J12934 — APPROVED** [review]

**Comment by J12934** [comment]
Merged and released with [v3.8.0](https://github.com/iteratec/multi-juicer/releases/tag/v3.8.0)

Thanks again, much appropriated 🙌

(I've added the [hacktoberfest-accepted](https://hacktoberfest.digitalocean.com/) label to the pr in case you are participating in that)

**Comment by DerGut** [comment]
Great, thanks! I started to extend the reset-password endpoint for the admin user too. However, I couldn't really find space in the admin table to display the new code 😄 
Perhaps we can track that in #63

**Pull request merged**